docs/administration/recent_activity/recent_activity.md

```diff
@@ -0,0 +1,231 @@
+  2024-01-29T14:58:53+00:00   317       “Foo Company Ltd.“   update         admin
+  2024-01-29T14:58:53+00:00   317       “Foo Company Ltd.“   publish        admin
+  2024-01-29T14:58:53+00:00   319       “Address Book“       create         admin
+  2024-01-29T14:58:53+00:00   319       “Address Book“       publish        admin
+  2024-01-29T14:58:53+00:00   317       “Foo Company Ltd.“   create_draft   admin
+  2024-01-29T14:58:53+00:00   317       “Foo Company Ltd.“   update         admin
+  2024-01-29T14:58:53+00:00   317       “Foo Company Ltd.“   publish        admin
+  2024-01-29T14:58:53+00:00   320       “HQ“                 create         admin
+  2024-01-29T14:58:53+00:00   320       “HQ“                 publish        admin
+  2024-01-29T14:58:53+00:00   317       “Foo Company Ltd.“   create_draft   admin
+  2024-01-29T14:58:53+00:00   317       “Foo Company Ltd.“   update         admin
+  2024-01-29T14:58:53+00:00   317       “Foo Company Ltd.“   publish        admin
+ --------------------------- --------- -------------------- -------------- -------
+```
+
+### Adding custom Activity Log entries
+
+!!! caution
+
+    Keep activity logging as light as possible. Do not make database request or heavy computation at logging time. Keep them for activity log list display time.
+
+#### Entry
+
+Your custom features can write into the activity log.
+
+First, inject `Ibexa\Contracts\ActivityLog\ActivityLogServiceInterface` into your PHP class having to log an activity (such as a custom event subscriber, event listener, service, or controller).
+
+In the following example, an event subscriber is subscribing to an event dispatched by a custom feature. This event has the information needed by a log entry (see details after the example).
+
+```php
+[[= include_file('code_samples/recent_activity/src/EventSubscriber/MyFeatureEventSubscriber.php') =]]
+```
+
+`ActivityLogService::build` function returns an `Ibexa\Contracts\ActivityLog\Values\CreateActivityLogStruct` which can then be passed to `ActivityLogService::save`.
+
+`ActivityLogService::build` has three arguments:
+
+* `$className` is the FQCN of the object actually manipulated by the feature. For example `Ibexa\Contracts\Core\Repository\Values\Content\Content::class`
+* `$id` is the ID or identifier of the manipulated object. For example, the Content ID cast to string.
+* `$action` is the identifier of the performed object manipulation. For example, `create`, `update` or `delete`.
+
+The returned `CreateActivityLogStruct` is always related to the currently logged-in user.
+
+If the object you log an activity on can become unavailable (like after a `delete` action), you might want to also log the name the object has at log time to be able to display it even when the object becomes unavailable. To add this name, use `CreateActivityLogStruct::setName` before saving the log entry.
+
+#### Context group
+
+If you log several entries at once, you can group them into a context.
```

Comment:
I'd mention that it's for _related entries_, that belong together as part of the same operation. Maybe with a short example.

Reply:
```suggestion
If you log several related entries at once, you can group them into a context.
```
Not sure about the necessity of a short example. Maybe this is the long example that should be better or earlier introduced.

Reply:
I really mean an example of what an activity log group is. Unless I have missed it, we do not explain the concept and what it is for. A couple examples, like creating a content item and maybe a migration or custom script, could be very useful.

But definitely no need for a code example.

Reply:
```suggestion
If you log several related entries at once, you can group them into a context.
A context is a set of actions done for the same purpose.
For example, a context could group the actions of CRON fetching third party data and updating Content items.
Some built-in contexts are:

- `web` to group the update and the publishing of a new Content item's version,
- or `migration` to group every actions from a migration file execution.
```